pkg/serviceprovider/snyk/snyk.go

```diff
@@ -0,0 +1,117 @@
+package snyk
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/serviceprovider"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/config"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ serviceprovider.ServiceProvider = (*Snyk)(nil)
+
+type Snyk struct {
+	Configuration config.Configuration
+	lookup        serviceprovider.GenericLookup
+	httpClient    rest.HTTPClient
+}
+
+var Initializer = serviceprovider.Initializer{
+	Probe:       snykProbe{},
+	Constructor: serviceprovider.ConstructorFunc(newSnyk),
+}
+
+func newSnyk(factory *serviceprovider.Factory, _ string) (serviceprovider.ServiceProvider, error) {
+
+	cache := serviceprovider.NewMetadataCache(factory.KubernetesClient, &serviceprovider.NeverMetadataExpirationPolicy{})
+
+	return &Snyk{
+		Configuration: factory.Configuration,
+		lookup: serviceprovider.GenericLookup{
+			ServiceProviderType: api.ServiceProviderTypeSnyk,
+			TokenFilter:         &tokenFilter{},
+			MetadataProvider: &metadataProvider{
+				httpClient:   factory.HttpClient,
+				tokenStorage: factory.TokenStorage,
+			},
+			MetadataCache: &cache,
+			RepoHostParser: serviceprovider.RepoHostParserFunc(func(repoUrl string) (string, error) {
```

Comment:
@metlos  is that loopback ok there ?

Reply:
This is used to pick all tokens that belong to the same host (i.e. all github tokens in a namespace, all quay tokens, etc.) so if this is enough to do that then it is ok.